### PR TITLE
Catch errors parseQueries

### PR DESCRIPTION
**Pull request by fogelito**
## What does this PR do?

(Provide a description of what this PR does and why it's needed.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

### CODE REVIEW

File: app/controllers/api/databases.php [hunk 1]
```diff
@@ -2912,7 +2912,13 @@ function updateAttribute(
             throw new Exception(Exception::COLLECTION_NOT_FOUND);
         }
 
-        $queries = Query::parseQueries($queries);
+        try {
+            $queries = Query::parseQueries($queries); // todo: make this to all  parseQueries places?
+        } catch (QueryException $e) {
+            // question: should this throw GENERAL_ARGUMENT_INVALID like all QueryException and get 500 ?
+            // or GENERAL_QUERY_INVALID and get 400?
```
Comment (abnegate) [inline app/controllers/api/databases.php:2919]: Let's use query invalid with a 400 response

File: app/controllers/api/databases.php [hunk 2]
```diff
@@ -2912,7 +2912,13 @@ function updateAttribute(
             throw new Exception(Exception::COLLECTION_NOT_FOUND);
         }
 
-        $queries = Query::parseQueries($queries);
+        try {
+            $queries = Query::parseQueries($queries); // todo: make this to all  parseQueries places?
```
Comment (abnegate) [inline app/controllers/api/databases.php:2916]: Yep let's do this everywhere

File: phpunit.xml [hunk 3]
```diff
@@ -6,7 +6,7 @@
          convertNoticesToExceptions="true"
          convertWarningsToExceptions="true"
          processIsolation="false"
-         stopOnFailure="false"
+         stopOnFailure="true"
```
Comment (abnegate) [inline phpunit.xml:9]: Let's revert

File: app/controllers/api/databases.php [hunk 4]
```diff
@@ -2933,14 +2939,13 @@ function updateAttribute(
...
 
-
         try {
             $documents = $dbForProject->find('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId(), $queries);
             $total = $dbForProject->count('database_' . $database->getInternalId() . '_collection_' . $collection->getInternalId(), $queries, APP_LIMIT_COUNT);
         } catch (AuthorizationException) {
             throw new Exception(Exception::USER_UNAUTHORIZED);
         } catch (QueryException $e) {
-            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, $e->getMessage());
+            throw new Exception(Exception::GENERAL_ARGUMENT_INVALID, $e->getMessage()); //  Should this be GENERAL_QUERY_INVALID? or 500 is ok?
```
Comment (abnegate) [inline app/controllers/api/databases.php:2948]: Let's use query invalid with a 400 response

File: app/controllers/api/databases.php (hunk 2 above)
Comment (fogelito) [reply app/controllers/api/databases.php]: Done!

File: app/controllers/api/databases.php (hunk 1 above)
Comment (fogelito) [reply app/controllers/api/databases.php]: Done!

File: phpunit.xml (hunk 3 above)
Comment (fogelito) [reply phpunit.xml]: Done!

File: app/controllers/api/databases.php (hunk 4 above)
Comment (fogelito) [reply app/controllers/api/databases.php]: Done!

**Review by abnegate — APPROVED** [review]

**Pull request merged**